src/tiled/objectselectionitem.cpp

```diff
@@ -229,13 +238,18 @@ void MapObjectLabel::updateColor()
 
 QRectF MapObjectLabel::boundingRect() const
 {
-    return mBoundingRect.adjusted(0, 0, 1, 1);
+    return mBoundingRect.translated(mScreenspaceOffset).adjusted(0, 0, 1, 1);
 }
 
 void MapObjectLabel::paint(QPainter *painter,
                            const QStyleOptionGraphicsItem *,
                            QWidget *)
 {
+    if (mObject->shape() == MapObject::Point) {
+        painter->save();
+        painter->translate(mScreenspaceOffset);
```

Comment:
@bjorn technically since we always have other shapes set to { 0, 0 } offset, we could remove the guards on `mObject->shape() == MapObject::Point`, but keeping the guards does save us from needlessly saving and restoring the painter transformation matrices.

Reply:
To make it a bit more future-proof, you could check `!mScreenspaceOffset.isNull()` instead of the object shape.

Reply:
@bjorn done - thanks!